### ISSUE TITLE
Feature-gate error when `CARGO_REGISTRIES_CRATES_IO_PROTOCOL` is set in the environment

### ISSUE THREAD

**Issue by jyn514** [willcrichton/flowistry#73]
# Problem

Flowistry is based off a toolchain from 1.67 (cargo +nightly-2022-12-07), which doesn't support sparse registries (they were stabilized in 1.68). Would it be possible to base this off a more recent toolchain instead?

# Logs

**OS:** darwin (22.4.0)
**VSCode:** 1.74.2
**Error message**
```
error: usage of sparse registries requires `-Z sparse-registry`

```

**Full log:** https://paste.rs/MFp

**Comment by jyn514** [comment]
As a workaround, manually running `cargo +nightly-2022-12-07 install flowistry_ide --version 0.5.35 --force -Z sparse-registry` gets flowistry far enough that it doesn't try to install the binary itself.

**Comment by jyn514** [comment]
ah no, that just delays the error 😞 
```
Flowistry could not run because your project failed to build with error:
error: failed to get `aho-corasick` as a dependency of package `sensor v1.9.3 (/Users/jyn/src/sensor)`

Caused by:
  failed to load source for dependency `aho-corasick`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  usage of sparse registries requires `-Z sparse-registry`
```

**Comment by willcrichton** [comment]
I will 0.5.36 soon which is on nightly-2023-04-12, is that recent enough?

**Comment by jyn514** [comment]
That should work great! 1.68 branched in January so April is more than late enough :) actually at that point I think this will be on by default so I won't need the environment variable.

**Comment by MrNossiom** [comment]
> I will 0.5.36 soon which is on nightly-2023-04-12, is that recent enough?

Do you know when v0.5.36 will be released ?

**Comment by willcrichton** [comment]
I just released 0.5.36, so this should be fixed now.
